addons/website_quote/controllers/main.py

```diff
@@ -97,60 +79,50 @@ def post(self, order_id, token, **post):
+        User = request.env.user.sudo()
         if 'body' in request.session and request.session.body:
-            request.registry.get('sale.order').message_post(cr, SUPERUSER_ID, order_id,
-                    body=request.session.body,
-                    type=type,
-                    subtype=subtype,
-                    author_id=user.partner_id.id,
-                    context=context,
-                    attachments=attachments
-                )
+            request.env['sale.order'].browse(order_id).sudo().message_post(body=request.session.body, type=type,
+                    subtype=subtype, author_id=User.partner_id.id, attachments=attachments)
             request.session.body = False
         return True
 
     @http.route(['/quote/update_line'], type='json', auth="public", website=True)
     def update(self, line_id, remove=False, unlink=False, order_id=None, token=None, **post):
-        order = request.registry.get('sale.order').browse(request.cr, SUPERUSER_ID, int(order_id))
+        order = request.env['sale.order'].sudo().browse(int(order_id))
         if token != order.access_token:
             return request.website.render('website.404')
-        if order.state not in ('draft','sent'):
+        if order.state not in ('draft', 'sent'):
             return False
-        line_id=int(line_id)
+        line_id = int(line_id)
         if unlink:
-            request.registry.get('sale.order.line').unlink(request.cr, SUPERUSER_ID, [line_id], context=request.context)
+            request.env['sale.order.line'].browse(line_id).sudo().unlink()
             return False
-        number=(remove and -1 or 1)
-
-        order_line_obj = request.registry.get('sale.order.line')
-        order_line_val = order_line_obj.read(request.cr, SUPERUSER_ID, [line_id], [], context=request.context)[0]
-        quantity = order_line_val['product_uom_qty'] + number
-        order_line_obj.write(request.cr, SUPERUSER_ID, [line_id], {'product_uom_qty': (quantity)}, context=request.context)
+        number = (remove and -1 or 1)
+        order_line = request.env['sale.order.line'].browse(line_id)
+        quantity = order_line.product_uom_qty + number
+        order_line.sudo().write({'product_uom_qty': (quantity)})
         return [str(quantity), str(order.amount_total)]
 
     @http.route(["/quote/template/<model('sale.quote.template'):quote>"], type='http', auth="user", website=True)
     def template_view(self, quote, **post):
-        values = { 'template': quote }
+        values = {'template': quote}
         return request.website.render('website_quote.so_template', values)
 
     @http.route(["/quote/add_line/<int:option_id>/<int:order_id>/<token>"], type='http', auth="public", website=True)
     def add(self, option_id, order_id, token, **post):
         vals = {}
-        order = request.registry.get('sale.order').browse(request.cr, SUPERUSER_ID, order_id)
+        order = request.env['sale.order'].sudo().browse(order_id)
```

Comment:
give order_sudo or ordersudo variable name to know order instance with superuser